contracts/Staking.sol

```diff
@@ -664,6 +678,20 @@ contract Staking is Pauser, Whitelist {
         emit DelegationUpdate(_valAddr, delAddr, validator.tokens, delegator.shares, -int256(_tokens));
     }
 
+    function _removeFrom(address _addr, address[] storage _addrArray) private {
+        uint256 lastIndex = _addrArray.length - 1;
+        for (uint256 i = 0; i < _addrArray.length; i++) {
+            if (_addrArray[i] == _addr) {
+                if (i < lastIndex) {
+                    _addrArray[i] = _addrArray[lastIndex];
+                }
+                _addrArray.pop();
```

Comment:
This may result in a large gas cost - my understanding of the code is going to copy/move the rest entries of _addrArray.

Reply:
Looks like my point is not yet addressed.  Have you estimated the gas cost if the array is long (since the array may be potentially very long)?

Reply:
Actually, it will only move once at most. About 280,000 gas cost for a list of 100 addresses in the worst case (hit the second last); about 50000 for 10.

Reply:
That still seems to be a lot, especially on Ethereum mainnet.  I am wondering if we could pass an `_index` parameter in the function (and callers) and therefore the removal can be done as
```
require(_addrArray[_index] == _addr || index = uint256(-1), "index not found");
if (_addrArray[_index] == _addr) {
  // copy the last and pop
} else {
  // search and pop
}
```
where _index can be computed efficiently off-chain.

Reply:
Draft it.